src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -76,42 +75,66 @@ public void openAndCloseTest() {
+        String allowedHeaderValChars = "!#$%&*+-.^_`~*|\"\'(){}[]<>=/,:;@\\" + alphaNumeric;
+        CompletableFuture<Void> httpRequestReceived = new CompletableFuture<Void>();
+        CompletableFuture<Void> connectionRequestReceived = new CompletableFuture<Void>();
+
+        listener.setRequestHandler(context -> {
+            try {
+                Map<String, String> headers = context.getRequest().getHeaders();
+                assertEquals(allowedHeaderValChars, headers.get(allowedHeaderNameChars));
+                context.getResponse().getHeaders().put(allowedHeaderNameChars, allowedHeaderValChars);
+                httpRequestReceived.complete(null);
+            } catch (Throwable ex) {
+                httpRequestReceived.completeExceptionally(ex);
+            } finally {
+                context.getResponse().close();
+            }
+        });
+        listener.setAcceptHandler(context -> {
+            try {
+                Map<String, String> headers = context.getRequest().getHeaders();
+                assertEquals(allowedHeaderValChars, headers.get(allowedHeaderNameChars));
+                connectionRequestReceived.complete(null);
+            } catch (Throwable ex) {
+                connectionRequestReceived.completeExceptionally(ex);
+            }
+            return true;
+        });
+        listener.openAsync(Duration.ofSeconds(10)).join();
+
+        // Test sending and receiving custom header on HybridConnection HTTP request from both client and listener
+        StringBuilder urlBuilder = new StringBuilder(TestUtil.RELAY_NAMESPACE_URI + TestUtil.ENTITY_PATH);
+        urlBuilder.replace(0, 5, "https://");
+        URL url = new URL(urlBuilder.toString());
+        String tokenString = tokenProvider.getTokenAsync(url.toString(), Duration.ofHours(1)).join().getToken();
+        
+        HttpURLConnection httpConnection = (HttpURLConnection)url.openConnection();
+        httpConnection.setRequestMethod("GET");
+        httpConnection.setRequestProperty(allowedHeaderNameChars, allowedHeaderValChars);
+        httpConnection.setRequestProperty("ServiceBusAuthorization", tokenString);
+        httpConnection.setDoOutput(true);
+        OutputStream out = httpConnection.getOutputStream();
+        out.close();
+        int responseCode = httpConnection.getResponseCode();
+
+        Map<String, List<String>> headers = httpConnection.getHeaderFields();
+        assertNotNull(headers);
+        assertEquals(httpConnection.getResponseMessage(), 200, responseCode);
+        httpRequestReceived.get(10, TimeUnit.SECONDS);
+
+        // Test sending custom header from client and receiving from listener for HybridConnection websocket mode
+        Map<String, List<String>> customHeaders = new HashMap<String, List<String>>();
+        customHeaders.put(allowedHeaderNameChars, Arrays.asList(new String[]{allowedHeaderValChars}));
+		CompletableFuture.allOf(client.createConnectionAsync(customHeaders).thenCompose(connection -> {
```

Comment:
>		 [](start = 0, length = 2)

Nit: The shows as not indented like the other surrounding lines in some editors because some lines use tabs and some lines use spaces.